### PR TITLE
Add support for mounting remote volumes using FTP

### PR DESCRIPTION
**Pull request by thallgren**
## Description

Adds an FTP server to the traffic-agent and a FUSE implementation based on an FTP-client to the user daemon's intercept mount logic. The feature is experimental at this stage and must be explicitly enabled using
a new boolean `intercept.useFtp` value in config.yml.

## Checklist

 - [x] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to add any docs changes required for my change (including release notes).
 - [x] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.
 - [ ] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [ ] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

### CODE REVIEW

File: pkg/client/userd/trafficmgr/intercept.go [hunk 1]
```diff
@@ -333,6 +450,34 @@ func (s *serviceProps) portIdentifier() (agentconfig.PortIdentifier, error) {
...
+			}
+		}
+		if ir.MountPoint != "" && iCept.ClientMountPoint == ir.MountPoint {
+			return &rpc.InterceptResult{
+				Error:         common.InterceptError_MOUNT_POINT_BUSY,
+				ErrorText:     spec.Name,
+				ErrorCategory: int32(errcat.User),
+				InterceptInfo: iCept.InterceptInfo,
+			}
+		}
```
Comment (josecv) [inline pkg/client/userd/trafficmgr/intercept.go]: can we make this a select please?

File: pkg/client/userd/trafficmgr/mount.go [hunk 2]
```diff
@@ -34,6 +52,41 @@ func (ic *intercept) startMount(ctx context.Context, podWG *sync.WaitGroup) {
...
+		host := fs.NewHost(m.client, ic.ClientMountPoint)
+		host.Start(ctx)
+		go func() {
+			// Stop when the intercept dies (but not when the pod changes)
+			<-ctx.Done()
+			defer dlog.Infof(ctx, "Unmounting FTP file system for intercept %q (address %s) at %q", ic.Id, addr, ic.ClientMountPoint)
+			host.Stop()
+		}()
+		return nil
+	}
```
Comment (josecv) [inline pkg/client/userd/trafficmgr/mount.go:83]: should we lock the client here?

**Review by josecv — CHANGES_REQUESTED** [review]
lgtm overall but some nits

File: pkg/client/userd/trafficmgr/intercept.go (hunk 1 above)
Comment (thallgren) [reply pkg/client/userd/trafficmgr/intercept.go]: No, but I can make it a switch ;-)

File: pkg/client/userd/trafficmgr/mount.go (hunk 2 above)
Comment (thallgren) [reply pkg/client/userd/trafficmgr/mount.go:83]: No, I don't think that's necessary. The `handleInterceptShapshot` method is dealing with intercepts sequentially, so there's no risk that the start method is called concurrently.

**Comment by thallgren** [comment]
Rebased on top of 2.7.3 and updated CHANGELOG.md to anticipate release in 2.7.4

**Review by josecv — APPROVED** [review]

**Pull request merged**